bootstrap: Update bootstrap plans for 9.2.8 to 9.12.2

mpickering opened the pull request:
- Removed 9.0.* plan
- Updated plans for 9.2.8, 9.4.8, 9.6.7, 9.8.4, 9.10.1, 9.12.2

Bumped the index state to get new enough versions to support 9.12.2.

Please read [Github PR Conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#github-pull-request-conventions) and then fill in *one* of these two templates.

---

**Template B: This PR does not modify behaviour or interface**

*E.g. the PR only touches documentation or tests, does refactorings, etc.*

Include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [] Is this a PR that fixes CI? If so, it will need to be backported to older cabal release branches (ask maintainers for directions).

ffaf1 reviewed (changes requested):
The recommended way to generate bootstraps is via `Makefile` ([C.3.5](https://github.com/haskell/cabal/wiki/Making-a-release#c3-bumping-version-numbers)).

How did you update those without editing the makefile?

ffaf1 commented:
I see @Mikolaj has pinged you Matrix:

> [Man of Letters] f-a: well spotted; the release checklist recommends the makefile as the way to generate bootstraps, so as long as it does, we should try to keep it operational and that probably requires an edit; I think the way :Matthew Pickering: updated the bootstraps is via Nix, which unfortunately requires Nix, so it's not a substitute for the makefile method[.]

mpickering commented:
> I see @Mikolaj has pinged you Matrix:
> 
> > [Man of Letters] f-a: well spotted; the release checklist recommends the makefile as the way to generate bootstraps, so as long as it does, we should try to keep it operational and that probably requires an edit; I think the way :Matthew Pickering: updated the bootstraps is via Nix, which unfortunately requires Nix, so it's not a substitute for the makefile method[.]

I updated the Makefile now with the versions which we have bootstrap plans for.
